tests/test_models_dbi_utils.py

```diff
@@ -42,13 +42,13 @@ def test_select_best_dbr_generator_and_run(backend, nqubits, step):
     initial_off_diagonal_norm = dbi.off_diagonal_norm
 
     for _ in range(NSTEPS):
-        idx, step_optimize = select_best_dbr_generator_and_run(
+        idx, step_optimize, flip_sign = select_best_dbr_generator_and_run(
```

Comment:
@Edoardo-Pedicillo When we have [D,H] then sometimes D-> (-D) is needed. Sam is working here with exploring all idx and finds this sign by the CS_angle function.

What is better for runtime: passing idx and then building D again by sign * "Pauli_ops[idx]" or passing back the operator e.g.
`D,s  =  select_best_dbr_generator_and_run(...)`?

Reply:
As suggested here https://github.com/qiboteam/qibo/pull/1143#discussion_r1447015892 `select_best_dbr_generator_and_run` should return the `dbi_object` with the best mode (i.e., with the best operator), `step_optimal` and `norms_off_diagonal_restriction`.

> What is better for runtime: passing idx and then building D again by sign * "Pauli_ops[idx]" or passing back the operator e.g.
D,s  =  select_best_dbr_generator_and_run(...)?

Since at the end you should always build D, I think the difference is negligible.